File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        d={}
        for i in nums:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        nums.clear()
        if 0 in d:
            while (d[0]):
                nums.append(0)
                d[0]-=1
        if 1 in d:
            while (d[1]):
                nums.append(1)
                d[1]-=1
        if 2 in d:
            while (d[2]):
                nums.append(2)
                d[2]-=1    
        

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low = 0; // Pointer for next position of 0
        int mid = 0; // Pointer for current element
        int high = nums.size() - 1; // Pointer for next position of 2

        while (mid <= high) {
            if (nums[mid] == 0) {
                swap(nums[low], nums[mid]);
                low++;
                mid++;
            } else if (nums[mid] == 1) {
                mid++;
            } else { // nums[mid] == 2
                swap(nums[mid], nums[high]);
                high--;
            }
        }
    }
};